cvat/apps/engine/serializers.py

```diff
@@ -405,9 +405,19 @@ def update_label(
             raise exceptions.ValidationError(str(exc)) from exc
 
         for attr in attributes:
-            (db_attr, created) = models.AttributeSpec.objects.get_or_create(
-                label=db_label, name=attr['name'], defaults=attr
-            )
+            attr_id = attr.get('id', None)
+            if attr_id is not None:
+                try:
+                    db_attr = models.AttributeSpec.objects.get(id=attr_id)
```

Comment:
Oh, sorry, I missed that we should also check that the attribute with the specified id belongs to the current label.
```suggestion
                    db_attr = models.AttributeSpec.objects.get(id=attr_id, label=db_label)
```

Reply:
Done